swap/protocol_test.go

```diff
@@ -315,3 +320,150 @@ func TestTriggerDisconnectThreshold(t *testing.T) {
 		t.Fatalf("Expected still no cheque, but there are %d", lenCheques)
 	}
 }
+
+// TestSwapRPC tests some basic things over RPC
+// We want this so that we can check the API works
+func TestSwapRPC(t *testing.T) {
+
+	var (
+		p2pPort = 30100
+		ipcPath = ".swarm.ipc"
+		err     error
+	)
+
+	swap, testDir := newTestSwap(t)
+	defer os.RemoveAll(testDir)
+
+	// need to have a dummy contract or the call will fail at `GetParams` due to `NewAPI`
+	swap.contract, err = contract.InstanceAt(common.Address{}, swap.backend)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// start a service stack
+	stack := createAndStartSvcNode(swap, ipcPath, p2pPort, t)
+	defer stack.Stop()
+
+	// connect to the servicenode RPCs
+	rpcclient, err := rpc.Dial(filepath.Join(stack.DataDir(), ipcPath))
+	if err != nil {
+		t.Fatal("connect to servicenode IPC fail", "err", err)
+	}
+	defer os.RemoveAll(stack.DataDir())
+
+	// create dummy peers so that we can artificially set balances and query
+	dummyPeer1 := newDummyPeer()
+	dummyPeer2 := newDummyPeer()
+	id1 := dummyPeer1.ID()
+	id2 := dummyPeer2.ID()
+
+	// set some fake balances
+	fake1 := int64(234)
```

Comment:
We can probably rename this to `fakeBalance1` or `peerBalance1` (something that includes a noun).